File: frontend/types/deal.ts
export interface Deal {
  id: number;
  title: string;
  value: number;
  stage: 'prospecting' | 'proposal' | 'negotiation' | 'closed_won' | 'closed_lost';
  contact_id: number | null;
  owner_id: number | null;
  expected_close_date: string | null;
  is_active: boolean;
  created_at: string;
}

export const STAGES = [
  { id: 'prospecting', label: 'Prospecting', color: '#94a3b8' },
  { id: 'proposal', label: 'Proposal', color: '#60a5fa' },
  { id: 'negotiation', label: 'Negotiation', color: '#fbbf24' },
  { id: 'closed_won', label: 'Closed Won', color: '#34d399' },
  { id: 'closed_lost', label: 'Closed Lost', color: '#f87171' },
] as const;

export interface DealContact {
  id: number;
  first_name: string;
  last_name: string;
  email: string | null;
  company_id: number | null;
  company_name: string | null;
}

export interface DealOwner {
  id: number;
  full_name: string;
}

export interface DealDetail extends Deal {
  contact: DealContact | null;
  owner: DealOwner | null;
}

File: backend/app/schemas/deal.py
from pydantic import BaseModel

from typing import Optional

from datetime import datetime

from enum import Enum

class DealStageEnum(str, Enum):

    prospecting = "prospecting"

    proposal = "proposal"

    negotiation = "negotiation"

    closed_won = "closed_won"

    closed_lost = "closed_lost"

class DealCreate(BaseModel):

    title: str

    value: Optional[float] = 0.0

    stage: DealStageEnum = DealStageEnum.prospecting

    contact_id: Optional[int] = None

    expected_close_date: Optional[datetime] = None

class DealUpdate(BaseModel):

    title: Optional[str] = None

    value: Optional[float] = None

    stage: Optional[DealStageEnum] = None

    contact_id: Optional[int] = None

    expected_close_date: Optional[datetime] = None

class DealResponse(BaseModel):
    id: int
    title: str
    value: float
    stage: DealStageEnum
    contact_id: Optional[int]
    owner_id: Optional[int]
    expected_close_date: Optional[datetime]
    is_active: bool
    created_at: datetime

    class Config:
        from_attributes = True


class DealContactSummary(BaseModel):
    id: int
    first_name: str
    last_name: str
    email: str
    company_id: Optional[int]
    company_name: Optional[str]


class DealOwnerSummary(BaseModel):
    id: int
    full_name: str


class DealDetailResponse(DealResponse):
    contact: Optional[DealContactSummary] = None
    owner: Optional[DealOwnerSummary] = None
class ActiveStatusUpdate(BaseModel):
    is_active: bool

    class Config:
        from_attributes = True